pennylane/operation.py

```diff
@@ -3034,6 +3035,57 @@ def convert_to_opmath(op):
     return op
 
 
+def convert_to_hamiltonian(op, wires_on_identity=True):
+    """
+    Converts arithmetic operators into :class:`~pennylane.Hamiltonian` instance.
+    Objects of any other type are returned directly.
+
+    Arithmetic operators include :class:`~pennylane.ops.op_math.Prod`,
+    :class:`~pennylane.ops.op_math.Sum` and :class:`~pennylane.ops.op_math.SProd`.
+
+    Args:
+        op (Operator): The operator instance to convert.
+        wires_on_identity (bool): If it is `True` preserves the wires associated with
+            the identity operator
+
+    Returns:
+        Operator: The operator as an :class:`~pennylane.Hamiltonian` instance
+    """
+    if isinstance(op, (qml.ops.op_math.Prod, qml.ops.op_math.SProd, qml.ops.op_math.Sum)):
+
+        op = qml.simplify(op)
+
+        if isinstance(op, Observable):
+            return qml.Hamiltonian(coeffs=[1], observables=[op])
+
+        identity_info = {}
+
+        # This is for tracking wires associated with identity
+        if wires_on_identity and isinstance(op, qml.ops.op_math.Sum):
+            for idx, o in enumerate(op):
+                if isinstance(o, qml.ops.op_math.SProd) and isinstance(o.base, qml.ops.Identity):
+                    identity_info["Id_idx"] = idx
+                    identity_info["Id_wires"] = o.base.wires
+                elif isinstance(o, qml.ops.Identity):
+                    identity_info["Id_idx"] = idx
+                    identity_info["Id_wires"] = o.wires
```

Comment:
Note that this is assuming just one Identity in the operator, but you can have a situation like the following (where the function currently has a bug)

```python
op = 0.7 * (I((0, 1)) + I(0))
H = convert_to_hamiltonian(op)
print(H.wires, H.coeffs, H.ops)
print(H)
```

```pycon
<Wires = [0, 1]> [1.4] [I(0)]
  (1.4) [I0,1]
```
The main problem here is that the operator wires dont match the `Wires.all_wires` of the obs